app/src/main/java/com/sdpteam/connectout/map/MapViewFragment.java

```diff
@@ -0,0 +1,76 @@
+package com.sdpteam.connectout.map;
+
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageButton;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.fragment.app.Fragment;
+import androidx.lifecycle.ViewModelProvider;
+
+import com.google.android.gms.maps.GoogleMap;
+import com.google.android.gms.maps.OnMapReadyCallback;
+import com.google.android.gms.maps.SupportMapFragment;
+import com.google.android.gms.maps.model.LatLng;
+import com.google.android.gms.maps.model.MarkerOptions;
+import com.sdpteam.connectout.R;
+import com.sdpteam.connectout.event.Event;
+
+import java.util.List;
+
+public class MapViewFragment extends Fragment implements OnMapReadyCallback {
+
+    private GoogleMap mMap;
+
+    private MapViewModel mapViewModel;
+
+    @Nullable
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+        View rootView = inflater.inflate(R.layout.fragment_map, container, false);
+
+        // Obtain the SupportMapFragment and get notified when the map is ready to be used.
+        //container object for the map used for the map lifecycle management and UI capabilities
+        SupportMapFragment mapFragment = (SupportMapFragment) getChildFragmentManager().findFragmentById(R.id.map);
+        mapFragment.getMapAsync(this);
+
+        mapViewModel = new ViewModelProvider(requireActivity()).get(MapViewModel.class);
+        mapViewModel.init(new MapModel());
+
+        mapViewModel.getEventList().observe(getViewLifecycleOwner(), eventList -> showNewMarkerList(eventList));
+
+        ImageButton refreshButton = rootView.findViewById(R.id.refresh_button);
+        refreshButton.setOnClickListener(view -> showNewMarkerList(mapViewModel.refreshEventList().getValue()));
+
+        return rootView;
+    }
+
+    private void showNewMarkerList(List<Event> eventList) {
+        if (mMap == null) {
+            return;
+        }
+        mMap.clear();
+        for (Event e : eventList) {
+            LatLng ll = new LatLng(e.getLat(), e.getLng());
```

Comment:
As I said before, following the principle of Tell-Don't-Ask, it would be preferable to have something like e.getPosition().toLatLng(), 
getPosition() should return a Position object that contains your GPS coordinates. And this object you can ask for converting to specific LatLng object for Google Maps stuff.